src/components/select/select.tsx

```diff
@@ -129,15 +143,32 @@ export class Select implements FormComponent {
   onSelectOptionDidLoad(ev: HTMLSelectOptionElementEvent) {
     const option = ev.target;
     if (this.value) {
+      // check if the select has a setted value
       if (this.value === option.value) {
+        // this select has a value and this
+        // option equals the correct select value
+        // so let's set this option as checked
         option.selected = true;
       } else {
+        // if the option value does not match
+        // with the select value,
+        // the option will be unchecked
+        // regardless if the option was
+        // initialized as checked
         option.selected = false;
       }
     } else {
+      // if the select does not have a value
+      // let's look for options initialized as checked
       if (option.selected) {
+        // this option was initialized as checked,
+        // so let's set the select's value
+        // equals to the checked option value
         this.value = option.value;
       }
+      // if no option is checked,
+      // all is unchecked and the select value
```

Comment:
perhaps here I can say something better... @ncardeli what do you think to comment here instead?

Reply:
@ncardeli 
maybe the comment could be something like:
```
      // If there is no option checked 
      // and no value was set in the select,
      // it will keep undefined until any 
      // change or checked option
```

Reply:
Yes, looks good

Reply:
this change was applied in f7924f0